src/HBL01MItineraryHashtag/HBL01MItineraryHashtag.service.ts

```diff
@@ -0,0 +1,84 @@
+import { Client } from '@elastic/elasticsearch';
+import { firestore } from 'firebase-admin';
+import { DocumentData, QueryDocumentSnapshot } from 'firebase-admin/firestore';
+
+import { HttpService } from '@nestjs/axios';
+import { Injectable } from '@nestjs/common';
+
+import { Itineraries } from 'spelieve-common/lib/Models/Itinerary/IDB01/Itineraries';
+
+import type { SearchResponse } from '@elastic/elasticsearch/lib/api/types';
+// import { PBL002UpsertPlaceDataServiceRule } from './PBL002UpsertPlaceData.rule';
+
+@Injectable()
+export class HBL01MItineraryHashtagService {
+  constructor(private readonly httpService: HttpService) {}
+
+  async doExecute() {
+    // リクエストに合致する QueryDocumentSnapshot を取得する
+    const itineraryCollectionRef = firestore().collection(
+      Itineraries.modelName,
+    );
+    const itineraryDocumentSnap: QueryDocumentSnapshot<DocumentData>[] | [] =
+      await itineraryCollectionRef
+        .get()
+        .then((qss) => (qss.empty ? [] : qss.docs));
+
+    const tag_list: Array<string> = itineraryDocumentSnap
+      .map((doc) => doc.data().tags)
+      .flat();
+
+    // count tags frequency
+    const toDict = (arr: Array<string>): { [key: string]: number } => {
+      const dict: { [key: string]: number } = {};
+      arr.map((e) => {
+        if (dict[e]) {
+          undefined += 1;
+        } else {
+          dict[e] = 1;
+        }
+      });
+      return dict;
+    };
+    const tagsDict: { [key: string]: number } = toDict(tag_list);
+
+    const indexName = 'search-m_itinerary_hashtags';
```

Comment:
spelieve-commonがもしかしたら更新されていないから更新しないとバグるかも

```suggestion
    const indexName = 'search-' + MItineraryHashtag.modelName;
```

Reply:
[1a8f5cd](https://github.com/Ayato-kosaka/spelieve-backend-tag/pull/3/commits/1a8f5cd1d048a6545e1b3598aa1bc7057d435dba)